clang/lib/AST/CanonBounds.cpp

```diff
@@ -310,7 +310,7 @@ Result Lexicographic::CompareExpr(const Expr *Arg1, const Expr *Arg2) {
 #include "clang/AST/StmtNodes.inc"
        llvm_unreachable("cannot compare a statement");  
      case Expr::PredefinedExprClass: Cmp = Compare<PredefinedExpr>(E1, E2); break;
-     case Expr::DeclRefExprClass: return Compare<DeclRefExpr>(E1, E2);
+     case Expr::DeclRefExprClass: Cmp = Compare<DeclRefExpr>(E1, E2); break;
```

Comment:
Why this change?

Reply:
Previously the code just returns here. This adds a new path so that if `Cmp` is not Equal, we proceed to use `CheckEquivExprs` to check if E1 and E2 produce the same value using EquivExprSets. (https://github.com/microsoft/checkedc-clang/pull/919/files#diff-9ff223384e8cae5a65b76ee59e13437eR376)

Reply:
Is there a test case that fails if this change is not made?

Reply:
```
void f() {
  int i1, i2;
  i1 = i2;
  _Array_ptr<int> p1 : count(i1 + 1) = 0;
  _Array_ptr<int> p2 : count(i2) = p1;
}
```

Without this change, `i1` and `i2` will be incorrectly reported as free variables in the code above, because`Compare<DeclRefExpr>` does not use the information from EquivExprSet to check if `i1` and `i2` can produce the same value. Once  `Compare<DeclRefExpr>` returns false, `CheckEquivExprs` also returns without using the information from the sets of equivalent expressions.

Previously in #903, we compare two ImplicitCastExprs rather than two DeclRefExprs, so this change was not needed. Now we directly compare DeclRefExprs collected from bounds expression, which is a different case for CheckEquivExprs.

Reply:
Thanks for clarifying!